site/docs/guide/deployment-types.md

```diff
@@ -257,3 +257,130 @@ const bot = new Bot(token, {
   },
 });
 ```
+
+This is how webhook replies work under the hood.
+
+```asciiart:no-line-numbers
+______________                                   _____________
+|            |                                   |           |
+|            |                                   |           |
+|            |                                   |           |
+|            |         *both waiting*            |           |
+|            |                                   |           |
+|  Telegram  |                                   |    Bot    |
+|            |                                   |           |
+|            |                                   |           |
+|            |    ---  hi, new message   --->    |           |
+|            | <--- okay, and sendChatAction --- |           |
+|____________|                                   |___________|
+```
+
+### Ending Webhook Requests in Time
+
+> You can ignore the rest of this page if all your middleware completes fast, i.e. within a few seconds.
+> This section is for example for people who want to do file transfers or other operations that need more time.
+
+When Telegram sends an update from one chat to your bot, it will wait for you to end the request before delivering the next update that belongs to that chat.
+In other words, Telegram will deliver updates from the same chat in sequence, and updates from different chats are sent concurrently.
+(The source of this information is [here](https://github.com/tdlib/telegram-bot-api/issues/75#issuecomment-755436496).)
+
+Telegram tries to make sure that your bot receives all updates.
+That means that if an update delivery fails for a chat, the subsequent updates will be queued until the first update succeeds.
+
+#### Why Not Ending a Webhook Request Is Dangerous
+
+Telegram has a timeout when sending an update to your webhook endpoint.
+If don't end a webhook request early enough, Telegram will resend the corresponding update, assuming it could not be delivered.
```

Comment:
Re-sending updates because of a timeout or an error are two different cases, and I'd like to keep the two things separate. Otherwise, I'd go with this:
```suggestion
If you don't end a webhook request fast enough, Telegram will re-send the update, assuming that it was not delivered.
```